akka-http/src/main/scala/akka/http/javadsl/server/Marshaller.scala

```diff
@@ -4,8 +4,108 @@
+  
+  def byteStringToEntity: Marshaller[ByteString,RequestEntity] = wrap(marshalling.Marshaller.ByteStringMarshaller)
+  
+  def fromDataToEntity: Marshaller[FormData,RequestEntity] = wrap(marshalling.Marshaller.FormDataMarshaller)
+  
+  def wrapEntity[A,C](f:function.BiFunction[ExecutionContext,C,A], m: Marshaller[A,_ <: RequestEntity], mediaType: MediaType): Marshaller[C,RequestEntity] = {
+    val scalaMarshaller = assumeScala(downcast(m, classOf[RequestEntity]).unwrap)
+    wrap(scalaMarshaller.wrapWithEC(mediaType) { ctx => c:C => f(ctx,c) } ) 
+  }
+
+  def wrapEntity[A,C](f:function.Function[C,A], m: Marshaller[A,_ <: RequestEntity], mediaType: MediaType): Marshaller[C,RequestEntity] = {
+    val scalaMarshaller = assumeScala(downcast(m, classOf[RequestEntity]).unwrap)
+    wrap(scalaMarshaller.wrap(mediaType)(f.apply)) 
+  }
+
+  def byteStringMarshaller(t: ContentType): Marshaller[ByteString, RequestEntity] = {
+    wrap(scaladsl.marshalling.Marshaller.byteStringMarshaller(t))
+  }
+  
+  def opaque[A,B](f: function.Function[A,B]): Marshaller[A, B] = {
+    wrap(scaladsl.marshalling.Marshaller.opaque[A,B] { a => f.apply(a) })
+  }
+  
+  def entityToOKResponse[A](m: Marshaller[A,_ <: RequestEntity]): Marshaller[A,HttpResponse] = {
+    wrap(marshalling.Marshaller.fromToEntityMarshaller[A]()(m.unwrap))
+  }
+  
+  def entityToResponse[A](status: StatusCode, m: Marshaller[A,_ <: RequestEntity]): Marshaller[A,HttpResponse] = {
+    wrap(marshalling.Marshaller.fromToEntityMarshaller[A](status)(m.unwrap))
+  }
+  
+  def entityToResponse[A](status: StatusCode, headers: java.lang.Iterable[HttpHeader], m: Marshaller[A,_ <: RequestEntity]): Marshaller[A,HttpResponse] = {
+    wrap(marshalling.Marshaller.fromToEntityMarshaller[A](status, headers.asScala.toVector)(m.unwrap))
+  }
+  
+  def entityToOKResponse[A](headers: java.lang.Iterable[HttpHeader], m: Marshaller[A,_ <: RequestEntity]): Marshaller[A,HttpResponse] = {
+    wrap(marshalling.Marshaller.fromToEntityMarshaller[A](headers = headers.asScala.toVector)(m.unwrap))
+  }
+  
+  def oneOf[A, B] (m1: Marshaller[A, B], m2: Marshaller[A, B]): Marshaller[A, B] = {
+    wrap(marshalling.Marshaller.oneOf(m1.unwrap, m2.unwrap))
+  }
+  
+  def oneOf[A, B] (m1: Marshaller[A, B], m2: Marshaller[A, B], m3: Marshaller[A, B]): Marshaller[A, B] = {
+    wrap(marshalling.Marshaller.oneOf(m1.unwrap, m2.unwrap, m3.unwrap))
+  }
+  
+  def oneOf[A, B] (m1: Marshaller[A, B], m2: Marshaller[A, B], m3: Marshaller[A, B], m4: Marshaller[A, B]): Marshaller[A, B] = {
+    wrap(marshalling.Marshaller.oneOf(m1.unwrap, m2.unwrap, m3.unwrap, m4.unwrap))
+  }
+
+  def oneOf[A, B] (m1: Marshaller[A, B], m2: Marshaller[A, B], m3: Marshaller[A, B], m4: Marshaller[A, B], m5: Marshaller[A, B]): Marshaller[A, B] = {
+    wrap(marshalling.Marshaller.oneOf(m1.unwrap, m2.unwrap, m3.unwrap, m4.unwrap, m5.unwrap))
+  }
+}
+
+case class Marshaller[A,B] private (implicit val unwrap: marshalling.Marshaller[A,B]) {
```

Comment:
It's the way to create a marshaller from Java, using the factory methods in the companion object. `implicit` can probably go away, I think it's a leftover from when I moved the implicit conversions to the single class.